ext/ddtrace.c

```diff
@@ -262,20 +263,24 @@ static void ddtrace_activate(void) {
     zend_hash_init(&DDTRACE_G(traced_spans), 8, unused, NULL, 0);
     zend_hash_init(&DDTRACE_G(tracestate_unknown_dd_keys), 8, unused, NULL, 0);
 
-    if (ddtrace_has_excluded_module == true) {
+    if (!DDTRACE_G(disable) && ddtrace_has_excluded_module == true) {
         DDTRACE_G(disable) = 2;
     }
 
     // ZAI config is always set up
     pthread_once(&dd_activate_once_control, dd_activate_once);
     zai_config_rinit();
 
+    if (!DDTRACE_G(disable) && get_global_DD_INSTRUMENTATION_TELEMETRY_ENABLED()) {
+        ddtrace_sidecar_ensure_active();
+    }
+
     zend_string *sampling_rules_file = get_DD_SPAN_SAMPLING_RULES_FILE();
     if (ZSTR_LEN(sampling_rules_file) > 0 && !zend_string_equals(get_global_DD_SPAN_SAMPLING_RULES_FILE(), sampling_rules_file)) {
         dd_save_sampling_rules_file_config(sampling_rules_file, PHP_INI_USER, PHP_INI_STAGE_RUNTIME);
     }
 
-    if (strcmp(sapi_module.name, "cli") == 0 && !get_DD_TRACE_CLI_ENABLED()) {
+    if (!DDTRACE_G(disable) && strcmp(sapi_module.name, "cli") == 0 && !get_DD_TRACE_CLI_ENABLED()) {
```

Comment:
I'm curious why this bit had to change?

Reply:
To not promote a DDTRACE_G(disable) being 0 to 2. 0 means generally disabled, and then sidecar won't start at all for example.

Reply:
Are the meanings of these values documented somewhere already? Maybe that would have been obvious if I worked on the tracer for my day to day, but if not, I recommend documenting this change or documenting these values somewhere.

Reply:
It's been boolean in the past until I needed distinction because of CLI_ENABLED, which is evaluated much later than the DDTRACE_G(disabled).

An enum was my thought as well.